cli/lsp/tsc.rs

```diff
@@ -4695,7 +4696,42 @@ fn op_script_names(state: &mut OpState) -> ScriptNames {
       .graph_imports_by_referrer(scope)
     {
       for specifier in specifiers {
-        script_names.insert(specifier.to_string());
+        let Ok(resolved) = state
+          .state_snapshot
+          .resolver
+          .as_cli_resolver(Some(scope))
+          .resolve(
+            specifier.as_str(),
+            scope,
```

Comment:
It might make more sense to use the referrer from `graph_imports_by_referrer()` here